### PR TITLE
BaSP-MR-108: Trainers and Classes: implement react hook forms

### PR DESCRIPTION
**Pull request by MatiasSauvanet**
-Implement react hooks form libreary to classes and trainers.
-Implement joi validations to inputs and selects for each entity.
WIP:
-Still needs implement consistence to the forms.

### CODE REVIEW

File: src/Components/Classes/Form/index.jsx [hunk 1]
```diff
@@ -23,6 +27,24 @@ const Form = () => {
     trainerId: '',
     slots: ''
   });
+  const schema = Joi.object({
+    day: Joi.string()
+      .valid('Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday')
+      .required(),
+    hour: Joi.string().required(),
```
Comment (SantiH97) [inline src/Components/Classes/Form/index.jsx:34]: Here you require a string but in the select you're passing a number

File: src/Components/Classes/Form/index.jsx [hunk 2]
```diff
@@ -246,30 +264,35 @@ const Form = () => {
...
 
-  const onChangeInput = (e) => {
-    setNewClass({
-      ...newClass,
-      [e.target.name]: e.target.value
-    });
-  };
+  // const onChangeInput = (e) => {
+  //   setNewClass({
+  //     ...newClass,
```
Comment (SantiH97) [inline src/Components/Classes/Form/index.jsx:275]: Remove comments once your're done with all corrections

File: src/Components/Classes/Form/index.jsx [hunk 3]
```diff
@@ -283,23 +306,20 @@ const Form = () => {
...
-    e.preventDefault();
+    console.log(data);
   };
   useEffect(() => {
     getActivities(dispatch);
     getTrainers(dispatch);
     id && getClassById(id);
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [dispatch]);
```
Comment (SantiH97) [inline src/Components/Classes/Form/index.jsx:322]: Im not sure this dependency is doing anything

**Comment by MatiasSauvanet** [comment]
[Classes & Trainers - Implement React Hook Form](https://trello.com/c/xGQXvMC6/108-classes-trainers-implement-react-hook-form)

**Comment by MatiasSauvanet** [comment]
[Trainers - Implement redux](https://trello.com/c/xxVyctjY/99-trainers-implement-redux)

**Review by SantiH97 — APPROVED** [review]
Looking good :)

**Review by ukiweskamp — APPROVED** [review]

**Pull request merged**